scripts/updateapilist.py

```diff
@@ -104,52 +137,45 @@ def generate_table_contents(clients: List[CloudClient]) -> List[str]:
     return content_rows + pypi_links
 
 
-REPO_METADATA_URL_FORMAT = (
-    "https://raw.githubusercontent.com/{repo_slug}/main/.repo-metadata.json"
-)
+def allowed_repo(repo) -> bool:
+    return (
+        repo[REPO_RESPONSE_KEY].startswith("googleapis/python-")
+        and repo[REPO_RESPONSE_KEY] not in REPO_EXCLUSION
+        and not repo[REPO_RESPONSE_KEY]
+    )
 
 
-def client_for_repo(repo_slug) -> Optional[CloudClient]:
-    url = REPO_METADATA_URL_FORMAT.format(repo_slug=repo_slug)
+def client_for_repo(repo_slug, is_split_repo: bool = True) -> Optional[CloudClient]:
+    path = SPLIT_REPO_PATH_FORMAT if is_split_repo else MONO_REPO_PATH_FORMAT
+    path = path.format(repo_slug=repo_slug)
+    url = f"{RAW_CONTENT_BASE_URL}/{path}/{REPO_METADATA_FILENAME}"
     response = requests.get(url)
     if response.status_code != requests.codes.ok:
         return
 
     return CloudClient(response.json())
 
-REPO_EXCLUSION = [
-    # core libraries
-    "googleapis/python-api-core",
-    "googleapis/python-cloud-core",
-    # proto only packages
-    "googleapis/python-org-policy",
-    "googleapis/python-os-config",
-    "googleapis/python-access-context-manager",
-    "googleapis/python-api-common-protos",
-    # testing utilities
-    "googleapis/python-test-utils",
-]
 
+def get_clients_batch_from_response_json(response_json, is_split_repo: bool = True) -> List[CloudClient]:
+    repos_key = REPO_RESPONSE_KEY if is_split_repo else PACKAGE_RESPONSE_KEY
+    return [client_for_repo(repo[repos_key], is_split_repo) for repo in response_json if (not is_split_repo or allowed_repo(repo))]
 
-def allowed_repo(repo) -> bool:
-    return (
-        repo["full_name"].startswith("googleapis/python-")
-        and repo["full_name"] not in REPO_EXCLUSION
-        and not repo["archived"]
-    )
 
+def mono_repo_clients(token: str) -> List[CloudClient]:
+    # all mono repo clients
+    url = f"{BASE_API}/repos/{MONO_REPO}/contents/packages"
+    headers = {'Authorization': f'token {token}'}
+    response = requests.get(url=url, headers=headers)
 
-def get_clients_batch_from_response_json(response_json) -> List[CloudClient]:
-    return [client_for_repo(repo["full_name"]) for repo in response_json if allowed_repo(repo)]
+    return get_clients_batch_from_response_json(response.json(), is_split_repo=False)
 
-def all_clients() -> List[CloudClient]:
-    clients = []
-    first_request = True
-    token = os.environ['GITHUB_TOKEN']
 
+def split_repo_clients(token: str) -> List[CloudClient]:
```

Comment:
I suggest re-architecting just a tad to not have have so many `if split_repo` in so many different places. Consider something like this:

```python
@dataclass
class Extractor:
    path_format: str
    response_key: str
split_repo_extractor = Extractor(path_format="{repo_slug}/main",
                                 response_key="name")
mono_repo_extractor = Extractor(path_format="googleapis/google-cloud-python/main/packages/{repo_slug}",
                                response_key="full_name")


def client_for_repo(repo_slug, extractor) -> Optional[CloudClient]:
    path = extractor.path_format
    # etc.

def get_clients_batch_from_response_json(response_json, extractor) -> List[CloudClient]:
    repos_key = extractor.response_key
    # etc.


def split_repo_clients():
    #...
    return get_clients_batch_from_response_json(repositories, split_repo_extractor)

def mono_repo_clients():
    #...
    return get_clients_batch_from_response_json(repositories, mono_repo_extractor)

```

If you want to get fancier (no need) you could make `get_clients_batch_from_response_json` and its callees be methods of `Extractor`, so that you could have

```python
def split_repo_clients():
    #...
    return split_repo_extractor.get_clients_batch_from_response_json(repositories)

def mono_repo_clients():
    #...
    return mono_repo_extractor.get_clients_batch_from_response_json(repositories)


```

Reply:
FYI https://docs.python.org/3/library/dataclasses.html